core/services/keeper/registry_synchronizer_core.go

```diff
@@ -117,28 +106,23 @@ func (rs *RegistrySynchronizer) Close() error {
 	return nil
 }
 
-func (rs *RegistrySynchronizer) OnNewLongestChain(ctx context.Context, head models.Head) {
-	select {
-	case rs.chHeads <- head:
-	default:
-	}
-}
-
 func (rs *RegistrySynchronizer) run() {
-	ticker := time.NewTicker(rs.interval)
+	syncTicker := time.NewTicker(rs.interval)
```

Comment:
This seems like a fine patch for this PR. Could we make a story to "process logs as soon as they arrive, rather than checking for logs in the mailroom every interval period". I think that will be a bit smoother. Should also make tests run faster. I think we can make that a follow up story though.